Fix crash in embedded mysql library when mysql_library_init() is invoked
with argc = 0, as is supposed to be a supported thing to do.
Patch from upstream bug #57931.


diff -Naur mysql-5.5.8.orig/libmysqld/lib_sql.cc mysql-5.5.8/libmysqld/lib_sql.cc
--- mysql-5.5.8.orig/libmysqld/lib_sql.cc	2010-12-03 12:58:26.000000000 -0500
+++ mysql-5.5.8/libmysqld/lib_sql.cc	2011-01-12 23:10:55.837471629 -0500
@@ -510,8 +510,8 @@
     return 1;
   defaults_argc= *argcp;
   defaults_argv= *argvp;
-  remaining_argc= argc;
-  remaining_argv= argv;
+  remaining_argc= *argcp;
+  remaining_argv= *argvp;
 
   /* Must be initialized early for comparison of options name */
   system_charset_info= &my_charset_utf8_general_ci;
